nodejs-server/src/echoServer.ts

```diff
@@ -0,0 +1,333 @@
+import * as grpc from 'grpc';
+import * as Long from 'long';
+
+import * as echoTypes from '../pbjs-genfiles/echo.js';
+
+import showcaseV1Beta1 = echoTypes.google.showcase.v1beta1;
+import google = echoTypes.google;
+import longrunning = google.longrunning;
+import { OperationsServer } from './operationsServer.js';
+
+/**
+ * Implements Echo server based on echo.proto.
+ */
+export class EchoServer {
+  private requestCount: number;
+  private paginationOutput: Map<string, string[]>;
+  private paginationRequest: Map<string, string>;
+  private operationsServer: OperationsServer;
+  private verbose: boolean;
+
+  constructor(verbose: boolean, operationsServer: OperationsServer) {
+    this.requestCount = 0;
+    this.paginationOutput = new Map<string, string[]>();
+    this.paginationRequest = new Map<string, string>();
+    this.operationsServer = operationsServer;
+    this.verbose = verbose;
+  }
+
+  private log(request: number, ...args: Array<string | {}>) {
+    if (this.verbose) {
+      console.log(`[EchoServer request #${request}]`, ...args);
+    }
+  }
+
+  /**
+   * Helper function to extract `google.rpc.Status` from some requests that
+   * contain this field.
+   */
+  private static requestToStatus(
+    request:
+      | showcaseV1Beta1.EchoRequest
```

Comment:
a bit of a meta question, about how we structure our TypeScript libraries in general. Will the default entry point of the library be whatever the current GA release of an API is. As an example, assume that we have API for a library that looks like this:

```
protos/google
   pubsub/v1
```

:point_up: when the library is first released, I assume our main `index.js` entry point would be the `pubsub/v1` version of the library ... now, 6 months passes, we release some amazing new features:

```
protos/google
   pubsub/v1
   pubsub/v2beta1
   pubsub/v2
```

Would the main entry point of the library become `pubsub/v2` once it's no longer in beta?